evalml/pipelines/components/transformers/dimensionality_reduction/lda.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as SkLDA
+
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils.gen_utils import is_all_numeric
+
+
+class LDA(Transformer):
+    """Reduces the number of features by using Linear Discriminant Analysis"""
+    name = 'Linear Discriminant Analysis Transformer'
+    hyperparameter_ranges = {}
+
+    def __init__(self, n_components=None, random_state=0, **kwargs):
+        """Initalizes an transformer that reduces the number of features using linear discriminant analysis."
+
+        Arguments:
+            n_components (int): the number of features to maintain after computationn. Defaults to None.
+        """
+        parameters = {"n_components": n_components}
```

Comment:
What happens if the user passes n_components > min(n_classes - 1, n_features)? Should we prevent that?

Reply:
Scikit-learn throws its own error here, it would probably make sense to catch it ourselves first!